bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -230,17 +232,19 @@ protected void removeModuleHandlerFactory(ModuleHandlerFactory moduleHandlerFact
      */
     @Override
     public Rule add(Rule rule) {
-        if (rule == null) {
-            throw new IllegalArgumentException("The added rule must not be null!");
-        }
         String rUID = rule.getUID();
         if (rUID == null) {
             rUID = ruleEngine.getUniqueId();
             super.add(initRuleId(rUID, rule));
         } else {
             super.add(rule);
         }
-        return get(rUID);
+        Rule ruleCopy = get(rUID);
```

Comment:
I don't have a worning in places where I use this method. Btw these warnings are related with missing null checks. Isn't be better to have handle these situations (as we saw that @Nonnull  does not defends us) and provider a suitable exception instead of just a NullPointerException at runtime. 
Anyway I can live with this change.